business/jaeger.go

```diff
@@ -99,7 +102,23 @@ func (in *JaegerService) GetAppTraces(ns, app string, query models.TracingQuery)
 	if err != nil {
 		return nil, err
 	}
-	return client.GetAppTraces(ns, app, query)
+	r, err := client.GetAppTraces(ns, app, query)
+	if err != nil {
+		return nil, err
+	}
+	if len(r.Data) == query.Limit {
```

Comment:
Ok, so, if I understand the implementation, first it fetches until reach the limit (i.e. 20 traces).
If limit is reached, then it performs the split and try to query 10 times over the given interval, dividing the limit (20 traces) with the slices (10).

Ok, so, in that case, when I have a query with 20 limit, it'll return 40 results.

Isn't it ?

Or perhaps we can just query the very same limit on the slices, and allow the user to personalize the slices as a parameter.

In any case, it sounds ok as a first iteration, I know it's not an easy topic.

Reply:
> Ok, so, if I understand the implementation, first it fetches until reach the limit (i.e. 20 traces).
> If limit is reached, then it performs the split and try to query 10 times over the given interval, dividing the limit (20 traces) with the slices (10).
> 
> Ok, so, in that case, when I have a query with 20 limit, it'll return 40 results.
> 
> Isn't it ?

Exactly

> 
> Or perhaps we can just query the very same limit on the slices, and allow the user to personalize the slices as a parameter.

Yes, perhaps. That should go with changes in the UI too, then. Perhaps we can think about that as part of our discussion about the toolbar redesign? ( here https://github.com/kiali/kiali/issues/2776 )

> 
> In any case, it sounds ok as a first iteration, I know it's not an easy topic.

Reply:
> Yes, perhaps. That should go with changes in the UI too, then. Perhaps we can think about that as part of our discussion about the toolbar redesign? ( here #2776 )

Totally, it makes sense.